common/JackControlAPI.cpp

```diff
@@ -712,6 +801,16 @@ jackctl_wait_signals(jackctl_sigmask_t * sigmask)
         // bugs that cause segfaults etc. during shutdown.
         sigprocmask(SIG_UNBLOCK, &sigmask->signals, 0);
     }
+
+#ifdef __linux__
+fail:
+    for(int i = 0; i < JackFDCount; i++) {
+        if(pfd[i].fd != 0) {
```

Comment:
This is wrong. If `eventfd()` fails then `close(-1)` gets called here. Initialize all fds to -1 and compare against that as the canonical "invalid fd" value.